pkg/service/group_service_factory.go

```diff
@@ -53,7 +55,7 @@ type NsGroupService struct {
 	witnessService                    *WitnessService
 	jobService                        *JobService
 	auditLogService                   *AuditLogService
-	poolService                       *PoolService
+	poolService                       servicefactory.PoolServiceHelper
```

Comment:
I think Vidhut is saying these need to be pointers.

Reply:
I was getting an error doing that. Since IVolumeService is a interface which contains a pointer to struct, we can use it directly? https://stackoverflow.com/questions/44370277/type-is-pointer-to-interface-not-interface-confusion